diff -Nur mapserver-5.6.8/mapraster.c mapserver-5.6.8.new/mapraster.c
--- mapserver-5.6.8/mapraster.c	2010-04-10 17:26:28.000000000 +0200
+++ mapserver-5.6.8.new/mapraster.c	2012-07-03 13:38:50.052734582 +0200
@@ -399,6 +399,28 @@
   return(0);
 }
 
+
+
+/*
+ * TIFF Image File Directories are comprised of a table of field
+ * descriptors of the form shown below.  The table is sorted in
+ * ascending order by tag.  The values associated with each entry are
+ * disjoint and may appear anywhere in the file (so long as they are
+ * placed on a word boundary).
+ *
+ * If the value is 4 bytes or less, then it is placed in the offset
+ * field to save space.  If the value is less than 4 bytes, it is
+ * left-justified in the offset field.
+ */
+typedef	struct {
+	uint16		tdir_tag;	/* see below */
+	uint16		tdir_type;	/* data type; see below */
+	uint32		tdir_count;	/* number of items; length in spec */
+	uint32		tdir_offset;	/* byte offset to field data */
+} TIFFDirEntry;
+ 
+
+
 /************************************************************************/
 /*                            readGEOTiff()                             */
 /*                                                                      */
